client/lib/analytics/page-view-tracker/index.jsx

```diff
@@ -1,8 +1,58 @@
-import analytics from 'lib/analytics';
-import PageViewTracker from './component';
+import React, { PropTypes } from 'react';
+import flowRight from 'lodash/flowRight';
+import has from 'lodash/has';
+import noop from 'lodash/noop';
+import { connect } from 'react-redux';
+import { recordPageView } from 'state/analytics/actions';
 
-const recorder = ( path, title ) => analytics.pageView.record( path, title );
+export const PageViewTracker = React.createClass( {
+	getInitialState: () => ( {
+		timer: null
+	} ),
 
-const Tracker = PageViewTracker( recorder );
+	componentDidMount() {
+		this.queuePageView();
+	},
 
-export default Tracker;
+	componentWillUnmount() {
+		clearTimeout( this.state.timer );
+	},
+
+	queuePageView() {
+		const {
+			delay = 0,
+			path,
+			recorder = noop,
+			title
+		} = this.props;
+
+		if ( this.state.timer ) {
+			return;
+		}
+
+		if ( ! delay ) {
+			return recorder( path, title );
+		}
+
+		this.setState( {
+			timer: setTimeout( () => recorder( path, title ), delay )
+		} );
+	},
+
+	render: () => null
+} );
+
+PageViewTracker.propTypes = {
+	delay: PropTypes.number,
+	path: PropTypes.string.isRequired,
+	recorder: PropTypes.func,
+	title: PropTypes.string.isRequired
+};
+
+const mapDispatchToProps = ( dispatch, props ) => ( {
+	recorder: has( props, 'recorder' )
+		? props.recorder
```

Comment:
I want to keep it because I can imagine overwriting it with a custom tracking function, especially for tests. In fact, if you look at the unit tests, that's exactly what I'm doing.

Do you think it's not likely though that outside of the tests we would be needing another one? I could be comfortable with that conclusion @Tug and then remove it from here, making this `mapDispatchToProps` a bit simpler